sdl-deltalake/src/main/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObject.scala

```diff
@@ -297,10 +310,16 @@ case class DeltaLakeTableDataObject(override val id: DataObjectId,
     implicit val helper: SparkSubFeed.type = SparkSubFeed
 
     val genericDf = SparkDataFrame(df)
-    val targetDf = saveModeOptions.map(_.convertToTargetSchema(genericDf)).getOrElse(genericDf).inner
-    val targetSchema = targetDf.schema
+    val targetDfIncoming = saveModeOptions.map(_.convertToTargetSchema(genericDf)).getOrElse(genericDf).inner
+    val targetSchema = targetDfIncoming.schema
+
+    val targetDf = if (schemaMin.isDefined) {
+      validateSchemaMin(SparkSchema(targetSchema), "write") //needed for merging the schemas
+      val sparkSchemaMin = schemaMin.get.asInstanceOf[SparkSchema] //writeSparkDataFrame is only done with SparkSubFeeds
+      val targetSchemaWithMetadata: StructType = SchemaUtil.mergeSchemaMetadata(sparkSchemaMin.inner, targetSchema)
+      session.createDataFrame(targetDfIncoming.rdd, targetSchemaWithMetadata); //workaround to replace the schema in the DF
```

Comment:
df.rdd triggers computations... better:
```suggestion
      targetDfIncoming.to(targetSchemaWithMetadata)
```